classes/classes/BodyParts/Skin.as

```diff
@@ -1,78 +1,145 @@
-package classes.BodyParts 
-{
-	/**
-	 * Container class for the players skin
-	 * @since December 27, 2016
-	 * @author Stadler76
-	 */
-	public class Skin 
-	{
-		include "../../../includes/appearanceDefs.as";
-
-		public var type:Number = SKIN_TYPE_PLAIN;
-		public var tone:String = "albino";
-		public var desc:String = "skin";
-		public var adj:String = "";
-		public var furColor:String = "no";
-
-		public function Skin() {}
-
-		public function skinFurScales():String
-		{
-			var skinzilla:String = "";
-			//Adjectives first!
-			if (adj != "")
-				skinzilla += adj + ", ";
+package classes.BodyParts {
+import classes.Creature;
+import classes.internals.Utils;
 
-			//Fur handled a little differently since it uses haircolor
-			skinzilla += hasFur() ? furColor : tone;
+/**
+ * Container class for the players skin
+ * @since December 27, 2016
+ * @author Stadler76
+ */
+public class Skin extends BodyPart {
+	include "../../../includes/appearanceDefs.as";
 
-			return skinzilla + " " + desc;
-		}
+	public var desc:String     = "skin";
+	public var furColor:String = "no";
+	public var tone:String = "albino";
+	public var adj:String = "";
 
-		public function description(noAdj:Boolean = false, noTone:Boolean = false):String
-		{
-			var skinzilla:String = "";
+	public function Skin(creature:Creature) {
+		super(creature);
```

Comment:
Skin doesn't use the Creature-field. I'd suggest to remove it from there. If your approach is flexible enough it won't break, if you simply remove it. Al least for now.